crates/database/src/settlements.rs

```diff
@@ -0,0 +1,85 @@
+use sqlx::PgConnection;
+
+use crate::TransactionHash;
+
+pub async fn recent_settlement_tx_hashes(
+    ex: &mut PgConnection,
+    start_block: i64,
+) -> Result<Vec<TransactionHash>, sqlx::Error> {
+    const QUERY: &str = r#"
+SELECT tx_hash
+FROM settlements
+WHERE block_number >= $1
```

Comment:
Should we also:
```suggestion
WHERE block_number >= $1 and block_number <= $deadline
```

I don't foresee this actually happening, but technically there could be a race condition where `self.current_block.borrow().number <= deadline` is true, but we are on block `deadline + 1` and a transaction indexed (for example if the future doesn't get scheduled for a while for whatever reason).

I'm not sure if this is actually possible in practice, but it might make sense to only accept transactions in the range.

Reply:
Is it a problem if we are accidentally more lenient here? I was considering this too but decided against it, because if we get the results of the query, we might as well use them.

Reply:
> Is it a problem if we are accidentally more lenient here?

I think it would only be an issue for debugging. I think the solver payout computation should happen with block chain data (and not count these solutions). However, if we make decisions based on success or failure (i.e. store a "success" flag in the database for example), then I don't think we should be lenient.

Reply:
Actually, given the discussion below, I think we don't want to make decisions about validity because of potential reorgs (which we can't reasonably deal with here).

With that in mind, the only issue I see is logging a "successfully mined" instead of "mined too late" which isn't a big deal.